### PR TITLE
Make authorization step part of the default payment gateway flow

### PR DESCRIPTION
**Pull request by patkleef**

### CODE REVIEW

File: src/Geta.Epi.Commerce.Payments.Netaxept.Checkout/Business/NetaxeptCheckoutPaymentGateway.cs [hunk 1]
```diff
@@ -115,7 +102,11 @@ public virtual PaymentProcessingResult ProcessPayment(IOrderGroup orderGroup, IP
                 return PaymentProcessingResult.CreateSuccessfulResult(stepResult.Message, stepResult.RedirectUrl);
             }
 
-            return PaymentProcessingResult.CreateUnsuccessfulResult(stepResult.Message);
+            // COM-10168 Payment: bug in Commerce 12 but fixed in Commerce 13.13. The unsuccessfull result is ignored by Episerver and payment is set to processed. When package
+            // is made compatible for Commerce 13 this code can be removed and the commented out code can be restored.
+            // return PaymentProcessingResult.CreateUnsuccessfulResult(stepResult.Message);
+
+            throw new Exception(stepResult.Message);
```
Comment (brianweet) [inline src/Geta.Epi.Commerce.Payments.Netaxept.Checkout/Business/NetaxeptCheckoutPaymentGateway.cs:109]: Can we throw a less generic exception here, like a payment exception?

File: src/Geta.Epi.Commerce.Payments.Netaxept.Checkout/Business/PaymentSteps/AuthorizationPaymentStep.cs [hunk 2]
```diff
@@ -1,112 +1,85 @@
...
-       /// <param name="payment"></param>
-       /// <param name="orderForm"></param>
-       /// <param name="orderGroup"></param>
-       /// <param name="message"></param>
-       /// <returns></returns>
-        public override PaymentStepResult Process(IPayment payment, IOrderForm orderForm, IOrderGroup orderGroup)
+        private PaymentStepResult HandlePaymentFailed(IOrderGroup orderGroup, IPayment payment, string title, string message)
         {
-            throw new NotImplementedException();
+            payment.Status = "Failed";
```
Comment (brianweet) [inline src/Geta.Epi.Commerce.Payments.Netaxept.Checkout/Business/PaymentSteps/AuthorizationPaymentStep.cs:19]: Can you change this to use PaymentStatus enum

**Comment by valdisiljuconoks** [comment]
Could there ever be cases that I would not want Authorize step to be part of default payment process?

**Comment by patkleef** [comment]
In the Netaxept flow, it's a required step.

**Pull request merged**